### PR TITLE
Merge contributing.md and dev_quick start docs

### PR DESCRIPTION
**Pull request by chatton**
This PR removes the dev_quick_start guide and includes all of the information in `Contributing.md`. 

This has been moved under the `Getting Started` section of the `Contributing.md` doc.

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -39,7 +39,14 @@ by the operator and mounted in the Agent's Pod.
 Each Pod holds a member of a Replica Set, and each Pod has different components,
 each one of them in charge of some part of the lifecycle of the MongoDB database.
 
-# Developing locally
+# Getting Started
```
Comment (bznein) [inline docs/contributing.md:42]: [q] do we have in this doc also a note about signing commits? We had a PR merged today in which the contributor had to rebase after setting up signing, we should put is as prerequisites to make the process smother

**Review by bznein — APPROVED** [review]
LGTM one q!

File: docs/contributing.md (hunk 1 above)
Comment (chatton) [reply docs/contributing.md:42]: Good point, will add something about it!

**Review by irajdeep — APPROVED** [review]
Nice one! LGTM

**Pull request merged**